contracts/FeralfileArtworkV4_1.sol

```diff
@@ -0,0 +1,147 @@
+        string memory symbol_,
+        bool burnable_,
+        bool bridgeable_,
+        address signer_,
+        address vault_,
+        address costReceiver_,
+        string memory contractURI_,
+        uint256[] memory seriesIds_,
+        uint256[] memory seriesMaxSupplies_,
+        AdvancedData[] memory data_
+    )
+        FeralfileExhibitionV4(
+            name_,
+            symbol_,
+            burnable_,
+            bridgeable_,
+            signer_,
+            vault_,
+            costReceiver_,
+            contractURI_,
+            seriesIds_,
+            seriesMaxSupplies_
+        )
+    {
+        for (uint256 i = 0; i < data_.length; i++) {
+            artistAdvancedAmounts[data_[i].artist] = data_[i].amount;
+        }
+    }
+
+    /// @notice pay to get artworks to a destination address. The pricing, costs and other details is included in the saleData
+    /// @param r_ - part of signature for validating parameters integrity
+    /// @param s_ - part of signature for validating parameters integrity
+    /// @param v_ - part of signature for validating parameters integrity
+    /// @param saleData_ - the sale data
+    function buyArtworks(
+        bytes32 r_,
+        bytes32 s_,
+        uint8 v_,
+        SaleData calldata saleData_
+    ) external payable override {
+        require(_selling, "FeralfileExhibitionV4: sale is not started");
+        super._checkContractOwnedToken();
+        validateSaleData(saleData_);
+
+        saleData_.payByVaultContract
+            ? vault.payForSale(r_, s_, v_, saleData_)
+            : require(
+                saleData_.price == msg.value,
+                "FeralfileExhibitionV4: invalid payment amount"
+            );
+
+        bytes32 message = keccak256(
+            abi.encode(block.chainid, address(this), saleData_)
+        );
+
+        require(
+            isValidSignature(message, r_, s_, v_),
+            "FeralfileExhibitionV4: invalid signature"
+        );
+
+        uint256 itemRevenue;
+        if (saleData_.price > saleData_.cost) {
+            itemRevenue =
+                (saleData_.price - saleData_.cost) /
+                saleData_.tokenIds.length;
+        }
+
+        uint256 distributedRevenue;
+        uint256 platformRevenue;
+        for (uint256 i = 0; i < saleData_.tokenIds.length; i++) {
+            // send NFT
+            _safeTransfer(
+                address(this),
+                saleData_.destination,
+                saleData_.tokenIds[i],
+                ""
+            );
+            // distribute royalty
+            RevenueShare[] memory revenueShares = saleData_.revenueShares[i];
+            bool isArtist = false;
+            address artistAddr;
+            for (uint256 j = 0; j < revenueShares.length; j++) {
+                if (artistAdvancedAmounts[revenueShares[j].recipient] > 0) {
+                    isArtist = true;
+                    artistAddr = revenueShares[j].recipient;
+                    break;
```

Comment:
This is the assumption that we only pay advance for one address. Since the royalty addresses passed as the `calldata`, so there is a chance multiple royalty addresses need to be paid the advance. In the contract, we should support that even the backend doesn't need. We also need to define the priority when setup the advance map so that we can rely on the priority to pay the advance in case there is multiple addresses need to be paid once. 
For example, we need to pay the advance for address A, B and C. The priorities are 1, 1 and 2. That means if we need to pay the advance, we should consider A and B as the first accounts to be paid then it goes to the account C.  That would be complicated but it will be more logical.